pkgs/development/python-modules/home-assistant-chip-wheels/default.nix

```diff
@@ -0,0 +1,267 @@
+{
+  appdirs,
+  appnope,
+  black,
+  build,
+  clang-tools,
+  click,
+  colorama,
+  coloredlogs,
+  coverage,
+  cryptography,
+  diskcache,
+  fetchFromGitHub,
+  glib,
+  gn1924,
+  googleapis-common-protos,
+  google-cloud-storage,
+  ipython,
+  jinja2,
+  json5,
+  jsonschema,
+  lark,
+  lib,
+  libnl,
+  mobly,
+  mypy,
+  mypy-extensions,
+  mypy-protobuf,
+  ninja,
+  packaging,
+  parameterized,
+  pip-tools,
+  pkg-config,
+  prompt-toolkit,
+  protobuf,
+  psutil,
+  ptpython,
+  pyelftools,
+  pygments,
+  pykwalify,
+  pylint,
+  pyperclip,
+  pyserial,
+  python3,
+  python-daemon,
+  pyyaml,
+  requests,
+  setuptools,
+  six,
+  sphinx,
+  sphinx-argparse,
+  sphinx-design,
+  stdenv,
+  stringcase,
+  toml,
+  tornado,
+  types-protobuf,
+  types-pyyaml,
+  types-requests,
+  types-setuptools,
+  watchdog,
+  websockets,
+  wheel,
+  yapf,
+  zap-chip,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "home-assistant-chip-wheels";
+  version = "2024.9.0";
+  src = fetchFromGitHub {
+    owner = "home-assistant-libs";
+    repo = "chip-wheels";
+    rev = version;
+    fetchSubmodules = true;
+    hash = "sha256-g+sBDO+i9uM+vYFdmeTgkhdGtQdcPSI9EHIvd+z19ro=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    gn1924
+    pkg-config
+    ninja
+    clang-tools
+    zap-chip
+    # gdbus-codegen
+    glib
+    python3
+    # dependencies of build scripts
+    click
+    jinja2
+    lark
+    setuptools
+    stringcase
+    build
+    pip-tools
+    black
```

Comment:
Could this fail with a black update? Can we easily avoid black to make formatter updates a no brainer?

Reply:
So there is a choice to make: 
- either we patch out dependencies which do no break the build. This would make the nix expression larger and even more fragile
- or we accept that there are more dependencies than actually needed, to keep the nix expression smaller.

current PR attempts to patch the least possible, except when dependencies are not in nixpkgs.

Reply:
Lets move this to the future when a black update breaks thing again or so